src/PowerShellEditorServices/Services/PowerShell/Execution/ExecutionOptions.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+
+namespace Microsoft.PowerShell.EditorServices.Services.PowerShell.Execution
+{
+    public enum TaskKind
+    {
+        Foreground,
+        Background,
+    }
+
+    public enum ExecutionPriority
+    {
+        Normal,
+        Next,
+    }
+
+    public record ExecutionOptions
+    {
+        public static ExecutionOptions Default = new()
+        {
+            Priority = ExecutionPriority.Normal,
+            MustRunInForeground = false,
+            InterruptCurrentForeground = false,
+        };
+
+        public ExecutionPriority Priority { get; init; }
+
+        public bool MustRunInForeground { get; init; }
+
+        public bool InterruptCurrentForeground { get; init; }
```

Comment:
I tried to work out what the simplest and most orthogonal set of options here was, and I'm still not really happy with it, but these two specifically are slightly different. One just prepends the next item, whereas the other will cancel the currently running thing.